### PR TITLE
docs: Update SDL and DDL documentation.

### PR DESCRIPTION
**Pull request by vpetrovykh**
The general description of SDL and DDL now have examples.

The link description now explains the possible "on delete" actions and
the DDL is updated accordingly.

### CODE REVIEW

File: docs/datamodel/links.rst [hunk 1]
```diff
@@ -24,3 +24,16 @@ The number of targets as specified by keywords :ref:`required
 and :ref:`multi <ref_eql_ddl_links_syntax>`.  It is also possible to
 restrict how many source objects can link to the same target via
 :eql:constraint:`exclusive` constraint.
+
+Links also have a policy of handling link target *deletion*. There are
+4 possible *actions* that can be taken when this happens:
+
+- ``RESTRICT`` - any attempt to delete the target object immediately
+  raises an exception
+- ``DELETE SOURCE`` - when the target of a link is deleted, also
```
Comment (elprans) [inline docs/datamodel/links.rst]: when the target of a link is deleted, the source is also deleted

File: docs/datamodel/links.rst [hunk 2]
```diff
@@ -24,3 +24,16 @@ The number of targets as specified by keywords :ref:`required
...
+  delete the source
+- ``ALLOW`` - the target object is deleted and is removed from the
+  set of the link targets
+- ``DEFERRED RESTRICT`` - any attempt to delete the target object
+  will raise an exception at the end of the transaction, unless by
```
Comment (elprans) [inline docs/datamodel/links.rst]: Please use consistent tense: *raises an exception*

File: docs/datamodel/links.rst (hunk 1 above)
Comment (vpetrovykh) [reply docs/datamodel/links.rst]: OK

File: docs/datamodel/links.rst (hunk 2 above)
Comment (vpetrovykh) [reply docs/datamodel/links.rst]: Fixed

File: docs/datamodel/links.rst [hunk 3]
```diff
@@ -24,3 +24,16 @@ The number of targets as specified by keywords :ref:`required
 and :ref:`multi <ref_eql_ddl_links_syntax>`.  It is also possible to
 restrict how many source objects can link to the same target via
 :eql:constraint:`exclusive` constraint.
+
+Links also have a policy of handling link target *deletion*. There are
+4 possible *actions* that can be taken when this happens:
+
+- ``RESTRICT`` - any attempt to delete the target object immediately
+  raises an exception
```
Comment (1st1) [inline docs/datamodel/links.rst]: Can you add ';' and finally a '.' in lists like this one?

File: docs/datamodel/links.rst [hunk 4]
```diff
@@ -24,3 +24,16 @@ The number of targets as specified by keywords :ref:`required
...
+- ``DELETE SOURCE`` - when the target of a link is deleted, the source
+  is also deleted
+- ``ALLOW`` - the target object is deleted and is removed from the
+  set of the link targets
+- ``DEFERRED RESTRICT`` - any attempt to delete the target object
+  raises an exception at the end of the transaction, unless by
+  that time this object is no longer in the set of link targets
```
Comment (1st1) [inline docs/datamodel/links.rst]: Need a link to DDL/SDL showing how to use this.

File: docs/datamodel/links.rst (hunk 3 above)
Comment (vpetrovykh) [reply docs/datamodel/links.rst]: OK

File: docs/datamodel/links.rst (hunk 4 above)
Comment (vpetrovykh) [reply docs/datamodel/links.rst]: Added.

**Pull request merged**